### PR TITLE
Obsolete `Pool` API cleansed

### PR DESCRIPTION
**Pull request by stankudrow**
Time to remove the obsolete code in the `Pool` module.

### CODE REVIEW

**Comment by stankudrow** [comment]
@DaniilAnichin , @itssimon, @DFilyushin, hello. You can have a look if you like, the API has got unburdened.

**Comment by DaniilAnichin** [comment]
The main question relevant to me, as `clickhouse-sqlalchemy[asynch]` user, is that versions of said library we're currently using do not have upper pin on the asynch (it is `>=0.2.2` in 0.3.1, and was changed to `>=0.2.2, <=0.2.4` in the 0.2.7, but 0.3.2 still has old pin), so I guess compatibility was broken in 0.2.5 already; Other than that, cleanup seems to be ok

**Comment by DaniilAnichin** [comment]
Also, there is another deprecation warning fired in the cursor tests in CI; should it be addressed, or we're testing to make sure old version still works?

**Review by DaniilAnichin — APPROVED** [review]

**Comment by stankudrow** [comment]
> Also, there is another deprecation warning fired in the cursor tests in CI; should it be addressed, or we're testing to make sure old version still works?

The one deprecation warning which leaves for the moment comes from cursors module and I didn't want to mix things into on melting PR.

**Comment by stankudrow** [comment]
> The main question relevant to me, as `clickhouse-sqlalchemy[asynch]` user, is that versions of said library we're currently using do not have upper pin on the asynch (it is `>=0.2.2` in 0.3.1, and was changed to `>=0.2.2, <=0.2.4` in the 0.2.7, but 0.3.2 still has old pin), so I guess compatibility was broken in 0.2.5 already; Other than that, cleanup seems to be ok

Sorry, I am not sure I understood the idea, could you rephrase what you meant?

**Comment by stankudrow** [comment]
@long2ice , ready to be merged.

**Comment by DaniilAnichin** [comment]
On the versioning and compatibility: sorry, I unfortunately tend to put stuff in such a way it's really hard to get
The idea is, I don't use / install asynch directly, so I'm first & foremost interested in `clickhouse-sqlalchemy` working as expected; 
And from the looks of it, this potential 0.2.6 release would be breaking, and will require pins in my code (nothing critical, but still)
So yes, I think it would be great to ensure proper upper pins on the `clickhouse-sqlalchemy` side; I will check your draft PR there about pyproject and poetry in the nearest future
